src/main/java/org/opentripplanner/api/common/RoutingResource.java

```diff
@@ -797,7 +798,11 @@ protected RouteRequest buildRequest(MultivaluedMap<String, String> queryParamete
           filterBuilder.addSelect(SelectRequest.of().withTransportModes(tModes).build());
         }
 
-        transit.setFilters(List.of(filterBuilder.build()));
+        if (!tModes.isEmpty()) {
+          transit.setFilters(List.of(filterBuilder.build()));
+        } else {
+          transit.setFilters(List.of(ExcludeAllTransitFilter.of()));
```

Comment:
We should not have it like this, as this runs the whole transit router, but the whole TransitRouter should be skipped instead

Reply:
This shortcut was removed in #4657 

We need to find a way to re-add it.

Reply:
https://github.com/opentripplanner/OpenTripPlanner/pull/4657#discussion_r1086641066